eth/fetcher/block_fetcher.go

```diff
@@ -754,7 +742,11 @@ func (f *BlockFetcher) importBlocks(peer string, block *types.Block) {
 		f.writeBlock(block)
 		// If import succeeded, broadcast the block
 		blockAnnounceOutTimer.UpdateSince(block.ReceivedAt)
-		go f.broadcastBlock(block, false)
+
+		// Only relay the Mined Blocks that meet the depth criteria
+		if relay {
+			go f.broadcastBlock(block, false)
```

Comment:
Why not propagate here?

Reply:
we propagate blocks to only  sqrt of blocks and broadcast hashes to all the peers